Parsers/ASimWebSession/Parsers/ASimWebSessionzScalerZIA.yaml

```diff
@@ -50,43 +53,48 @@ ParserQuery: |
     DstAppName = DestinationServiceName,
     DstIpAddr = DestinationIP,
     DstFQDN = DestinationHostName,
-    DstBytes = ReceivedBytes,
     SrcIpAddr = SourceIP,
     SrcUsername = SourceUserName,
     SrcNatIpAddr= SourceTranslatedAddress,
     SrcUserDepartment = SourceUserPrivileges, // Not part of the standard schema
-    SrcBytes = SentBytes,
     ThreatRiskLevel = DeviceCustomNumber1,
     UrlCategory = DeviceCustomString2,
     ThreatName = DeviceCustomString5,
     FileMD5 = DeviceCustomString6
   // -- Parse
   | parse AdditionalExtensions with 
       "reason=" EventResultOriginalDetails:string ";"
-      "outcome=" EventResultDetails:int ";"
+      "outcome=" EventResultDetails:string ";"
       "cat=" * ";"
       "rulelabel=" RuleName:string ";"
       "ruletype=" ruletype:string ";"
       "urlclass=" urlclass:string ";"
       "devicemodel=" * 
   // -- Calculated fields
+  | lookup DvcActionLookup on DeviceAction
   | extend
+    DvcHostname = tostring(Computer),
+    SrcBytes = toint(SentBytes),
+    DstBytes = toint(ReceivedBytes),
     Url = iff (RequestURL == "", "", strcat (tolower(NetworkApplicationProtocol), "://", url_decode(RequestURL))),
     UrlCategory = strcat (urlclass, "/", UrlCategory),
     ThreatCategory = iff(DeviceCustomString4 == "None", "", strcat (DeviceCustomString3, "/", DeviceCustomString4)),
     RuleName = iff (RuleName == "None", "", strcat (ruletype, "/", RuleName)),
     FileMD5 = iff (FileMD5 == "None", "", FileMD5),
     HttpReferrer = iff (RequestContext == "None", "", url_decode(RequestContext)),
     DstAppName = iff (DstAppName == "General Browsing", "", DstAppName),
-    DstFQDNparts = split (DstFQDN, ".")
+    DstFQDNparts = split (DstFQDN, "."),
+    DstHostnameNotAddr = DstIpAddr != DstFQDN
   | extend
-    DstHostname = DstFQDNparts[0],
-    DstDomain = strcat_array(array_slice(DstFQDNparts,1,-1),".")
+    DstHostname = iff (DstHostnameNotAddr, tostring(DstFQDNparts[0]), DstFQDN),
+    DstDomain = iff (DstHostnameNotAddr, strcat_array(array_slice(DstFQDNparts,1,-1),"."), ""),
+    DstFQDN = iff (DstHostnameNotAddr, DstFQDN, "") 
   // -- Enrichment
   | extend
     EventResult = iff (EventResultDetails == "NA" or toint(EventResultDetails) >= 400, "Failure", "Success"),
+    EventSeverity = case (ThreatRiskLevel > 90, "High", ThreatRiskLevel > 60, "Medium", ThreatRiskLevel > 10, "Low",  "Informational"),
     DstAppType = "SaaS application",
-    DstFQDN = "FQDN",
+    DstDomainType = iff (DstHostnameNotAddr, "FQDN", ""),
```

Comment:
Maybe we are missing a DomainType="IP"?

Reply:
Actually, I don't set a domain if the hostname is an IP.